### ISSUE TITLE
[REQUEST] Ability to "lock" threads for special usage

### ISSUE THREAD

**Issue by Micro-Crackedto** [openbullet/openbullet#433]
**Describe the solution you'd like**
I'd like an option in OpenBullet to lock threads to write to files and use browsers.
Basically I want OpenBullet to be able to write to a .txt with 1 thread at a time and not multiple as that will mess up the .txt
Same deal with the browser, would be nice to only have 1 chrome open when cracking with lets say 100+ threads.

**Describe alternatives you've considered**
I could create a .txt for each cracked account, in example: test@test.com.txt but this will create a bunch of .txt;s and make it unorganized.

**Comment by openbullet** [comment]
I need to implement a lock to not allow multiple threads to write the same file, but for now you can get around this by doing exactly what you're talking about. It's not hard to set up something like that, just keep the thread with BOTNUM equal to 1 (or 0 if debugging) as reserved for I/O operations or selenium browsing. I suggest you read this https://openbullet.github.io/examples.html it's the same concept.
Basically you would want to do:
```
IF "<BOTNUM>" EqualTo "1"
## Do single thread stuff

ELSE
## Do multithread stuff

ENDIF
```

I will leave this open so that I remember to implement the I/O lock.

**Comment by Micro-Crackedto** [comment]
Ah, so <BOTNUM> its not the total amount of bots being used, its just being "reset" whenever bot 1 is done and tries another account?

**Comment by openbullet** [comment]
The BOTNUM variable is the ID of the current bot that is handling that input data. For example in the Runner you can see the ID on the leftmost column, that's the same of what you would get in BOTNUM. For the debugger, it's always 0.

**Comment by Micro-Crackedto** [comment]
Right, but 

> IF "<BOTNUM>" EqualTo "1"

 will only works if i run 1 bot, so why the need of the code ?

**Comment by openbullet** [comment]
Because there is an ELSE statement after that, so everything in the ELSE statement will be run by bot 2 to 200 ^_^

**Comment by Micro-Crackedto** [comment]
Yeah but this wont actually prevent it from writing at the same time, or using selenium at the same time

**Comment by openbullet** [comment]
If you delegate the I/O operations only to bot number 1 (and other bots communicate the data to write through global variables) you actually can. But if you want to actually write on the same file using multiple threads, then I need to fix the problem that I described above.

**Comment by Micro-Crackedto** [comment]
Ah, so only bot #1 will do the writing of the global variables being used by other bots ?

**Comment by openbullet** [comment]
I understand your issue, just wait for me to add a lock, do not use the global variables approach as it would require a lot of work.

**Comment by openbullet** [comment]
This issue is probably a duplicate of https://github.com/openbullet/openbullet/issues/409 so I'm closing this one
